Documentation for columnar. (#151739)
elastic/elasticsearch: docs/reference/elasticsearch/mapping-reference/mapping-source-field.md

* [`version`](/reference/elasticsearch/mapping-reference/version.md#version-synthetic-source)
* [`wildcard`](/reference/elasticsearch/mapping-reference/keyword.md#wildcard-synthetic-source)

## Columnar source [columnar-stored]

```{applies_to}

> stack: preview

Reviewer: ```suggestion
stack: preview 9.5+
```

serverless: preview
```

> The columnar index modes use columnar source. By default this gets generated on the fly from doc values at query time.

Reviewer: "on the fly" seems like a phrase that wouldn't translate well
```suggestion
The columnar index modes use columnar source. By default this content gets generated dynamically from doc values at query time.
```

> The columnar index modes also support columnar source. By default this gets generated on the fly from doc values at query time.

Reviewer 2: ```suggestion
The columnar index modes use columnar source. By default this gets generated on the fly from doc values at query time.
```

> But this can also be stored on disk at index time by using `columnar_stored` source mode.

Reviewer: Ditto previous suggestion, it's unclear what "this" is. Content? Information? Data?
```suggestion
But this content can also be stored on disk at index time by using `columnar_stored` source mode.
```

The `columnar_stored` source mode can be useful for queries that fetch all or most of the fields in the index.

To use columnar-stored source:

```console
PUT my-columnar-index
{
  "settings": {
    "index": {
      "mode": "columnar"
    }
  },
  "mappings": {
    "_source": {
      "mode": "columnar_stored"
    }
  }
}
```

> Disabling `_source` entirely (`"_source": {"enabled": false}`) is **not allowed** in columnar modes.

Reviewer: ```suggestion
Turning off `_source` entirely (`"_source": {"enabled": false}`) is **not allowed** in columnar modes.
```

> Disabling `_source` entirely (`"_source": {"enabled": false}`) is **not allowed** in columnar modes.

Reviewer: ```suggestion
Turning off `_source` entirely (`"_source": {"enabled": false}`) is **not allowed** in columnar modes.
```

> Disabling `_source` entirely (`"_source": {"enabled": false}`) is **not allowed** in columnar modes.

Reviewer: One more minor suggestion re word choice (https://www.elastic.co/docs/contribute-docs/style-guide/word-choice)
```suggestion
Turning off `_source` entirely (`"_source": {"enabled": false}`) is **not allowed** in columnar modes.
```

> ## Disabling the `_source` field [disable-source-field]

Reviewer: ```suggestion
## Turning off the `_source` field [disable-source-field]
```
